pkg/client/client.go

```diff
@@ -0,0 +1,195 @@
+	}
+	var data map[string]interface{}
+	err = json.Unmarshal(respBody, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func Read(ctx context.Context, c *http.Client, serverUrl string, path string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating post request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var data map[string]interface{}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func Delete(ctx context.Context, c *http.Client, serverUrl string, path string) error {
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return fmt.Errorf("error creating delete request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	var data map[string]interface{}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func Update(ctx context.Context, c *http.Client, serverUrl string, path string, body map[string]interface{}) error {
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	reqBody, err := json.Marshal(body)
+	if err != nil {
+		return fmt.Errorf("error marshalling JSON for request body: %v", err)
+	}
+
+	req, err := http.NewRequest("PATCH", url, strings.NewReader(string(reqBody)))
+	if err != nil {
+		return fmt.Errorf("error creating delete request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+
+	var data map[string]interface{}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(respBody, &data)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return err
+	}
+
+	return CheckErrors(data)
+}
+
+func CheckErrors(resp map[string]interface{}) error {
+	e, ok := resp["error"]
+	if ok {
+		return fmt.Errorf("returned errors, %v", e)
+	}
+}
+
+func createBase(ctx context.Context, r *api.Resource, serverUrl string, parameters map[string]string, suffix string) string {
```

Comment:
```suggestion
func basePath(ctx context.Context, r *api.Resource, serverUrl string, parameters map[string]string, suffix string) string {
```

nit mostly - but `createBase` doesn't really follow go idioms I've seen where you're more stating the intended output, vs adding a superfluous verb.

Also this avoids people getting confused about `create` referring to the standard method.